src/components/pager/pager.test.js

```diff
@@ -589,4 +484,266 @@ context("Test for Pager component", () => {
+      (variant) => {
+        CypressMountWithProviders(<PagerComponent variant={variant} />);
+        cy.checkAccessibility();
+      }
+    );
+
+    it.each([true, false])(
+      "pager links are rendered as intended when hideDisabledElements is set to %s and currentPage is 1 for accessibility tests",
+      (boolVal) => {
+        CypressMountWithProviders(
+          <PagerComponent currentPage={1} hideDisabledElements={boolVal} />
+        );
+        cy.checkAccessibility();
+      }
+    );
+
+    it.each([true, false])(
+      "pager links are rendered as intended when hideDisabledElements is set to %s and currentPage is 10 for accessibility tests",
+      (boolVal) => {
+        CypressMountWithProviders(
+          <PagerComponent currentPage={10} hideDisabledElements={boolVal} />
+        );
+        cy.checkAccessibility();
+      }
+    );
+
+    it("both pager links are rendered when hideDisabledElements is set to true, but currenPage is above 1 for accessibility tests", () => {
+      CypressMountWithProviders(
+        <PagerComponent currentPage={7} hideDisabledElements />
+      );
+      cy.checkAccessibility();
+    });
+
+    describe("check funtionality for Pager component for accessibility tests", () => {
+      it.each([-1, -10, -100, testData[0], testData[1]])(
+        "should set totalRecords out of scope to %s for accessibility tests",
+        (totalRecords) => {
+          CypressMountWithProviders(
+            <PagerComponent totalRecords={totalRecords} />
+          );
+          cy.checkAccessibility();
+        }
+      );
+
+      it("should disable nextArrow and lastArrow buttons after clicking on lastArrow button for accessibility tests", () => {
+        CypressMountWithProviders(<PagerComponent currentPage={3} />);
+        cy.checkAccessibility();
+      });
+
+      it("should disable firstArrow and previousArrow buttons after clicking on firstArrow button for accessibility tests", () => {
+        CypressMountWithProviders(<PagerComponent currentPage={3} />);
+        cy.checkAccessibility();
+      });
+
+      it("should disable firstArrow and previousArrow buttons after clicking on previousArrow button for accessibility tests", () => {
+        CypressMountWithProviders(<PagerComponent currentPage={2} />);
+        cy.checkAccessibility();
+      });
+
+      it("should disable firstArrow and previousArrow buttons after clicking on nextArrow button for accessibility tests", () => {
+        CypressMountWithProviders(<PagerComponent currentPage={9} />);
+        cy.checkAccessibility();
+      });
+
+      it.each([1001, 901, 701, 601, 45])(
+        "should render Pager in %s px width",
+        (viewportWidth) => {
+          cy.viewport(viewportWidth, 768);
+
+          CypressMountWithProviders(<PagerComponentResponsive />);
+          cy.checkAccessibility();
+        }
+      );
+    });
+
+    describe("check Pager events for accessibility", () => {
```

Comment:
I don't think these event tests need to be converted to accessibility tests. Nothing is actually being triggered in the test, so you're just testing the default component repeatedly.

Reply:
+1